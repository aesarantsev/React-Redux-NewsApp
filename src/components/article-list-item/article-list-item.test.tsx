import * as React from "react";
import * as renderer from "react-test-renderer";
import ArticleListItem from "./";
import { BrowserRouter as Router } from "react-router-dom";

it("SubjectToBeTested renders correctly", () => {
  const props = {
    article: {
      id: "123123",
      title: "Test news title",
      description: "Test description",
      urlToImage: "https://cdn.vox-cdn.com/thumbor/SLLGigPOOcRWZOVySx01jGGI2sI=/0x109:1520x905/fit-in/1200x630/cdn.vox-cdn.com/uploads/chorus_asset/file/19719342/vbucks.0__1_.png",
      publishedAt: "2020-02-14T03:15:11Z",
      content: "test content"
    }
  };
  const tree = renderer
    .create(
      <Router>
        <ArticleListItem {...props} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
