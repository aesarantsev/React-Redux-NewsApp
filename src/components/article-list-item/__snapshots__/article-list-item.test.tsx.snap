// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubjectToBeTested renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <a
      href="/undefined/articles/123123"
      onClick={[Function]}
      style={
        Object {
          "color": "initial",
          "textDecoration": "none",
        }
      }
    >
      <img
        className="MuiCardMedia-root makeStyles-media-2 MuiCardMedia-media MuiCardMedia-img"
        src="https://cdn.vox-cdn.com/thumbor/SLLGigPOOcRWZOVySx01jGGI2sI=/0x109:1520x905/fit-in/1200x630/cdn.vox-cdn.com/uploads/chorus_asset/file/19719342/vbucks.0__1_.png"
      />
      <div
        className="MuiCardContent-root"
      >
        <h2
          className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
        >
          Test news title
        </h2>
        <p
          className="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary"
        >
          14 February 2020
        </p>
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          Test description
        </p>
      </div>
    </a>
    <span
      className="MuiCardActionArea-focusHighlight"
    />
  </button>
</div>
`;
